### PR TITLE
Update README with some style pointers and to remove outdated BCPC references

### PR DESCRIPTION
**Pull request by cbaenziger**
This tries to provide some ideas for how we should code and cleans up BCPC-isms in the README. This doesn't yet propose bringing our code into compliance with the style guides though.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,7 +145,6 @@ BCPC currently relies upon a number of open-source packages:
  - [Apache Oozie](http://oozie.apache.org/)
  - [Apache Pig](http://pig.apache.org/)
  - [Apache Sqoop](http://sqoop.apache.org/)
```
Comment (kiiranh) [inline README.md]: Apache Spark?

**Comment by cbaenziger** [comment]
Added Spark, Hannibal and HDFS-DU; corrected one errant BCPC too.

File: README.md [hunk 2]
```diff
@@ -75,7 +60,7 @@ section for enrolling the nodes.
 Make a cluster
 --------------
 
-To build a new BCPC cluster, you have to start with building a head nodes
+To build a new BACH cluster, you have to start with building a head nodes
```
Comment (mlongob) [inline README.md]: Grammar? 'a head node' or 'head nodes'?

Comment (cbaenziger) [inline README.md]: Thanks, correction pushed

File: README.md [hunk 3]
```diff
@@ -95,45 +80,64 @@ one can run through what is the expected "happy-path" install. This simple
...
 -------------
 
-BCPC currently relies upon a number of open-source packages:
+BACH currently relies upon a number of open-source packages:
 
- - [389 Directory Server](http://directory.fedoraproject.org/)
  - [Apache Bigtop](http://bigtop.apache.org/)
  - [Apache Hadoop](http://hadoop.apache.org/)
+ - [Apache Spark](http://spark.apache.org/)
  - [Apache HBase](http://hbase.apache.org/)
```
Comment (mlongob) [inline README.md]: Apache Phoenix, Apache Tez, Apache Zookeeper, MIT Kerberos ?

**Comment by mlongob** [comment]
Suggested some more open source tools. Rest looks good.

File: README.md (hunk 3 above)
Comment (cbaenziger) [inline README.md]: I added Phoenix, Tez and Zookeeper but we do not have an HA MIT Kerberos KDC in our stack (only a single node for integration testing) so I am not inclined to call that out yet.

**Comment by mlongob** [comment]
LGTM

**Comment by cbaenziger** [comment]
Thanks Mario!

**Pull request merged**